### ISSUE TITLE
Dart VM crashes during debugging when stepping into a function that references a constant variable (with a specific type of initializer))

### ISSUE THREAD

**Issue by boasjls** [dart-lang/sdk#24323]
## Summary:
The DartSDK crashes during debugging when stepping into a function that references a constant variable with a very particular type of initializer (refer to step 2).

## What steps will reproduce the problem?
1. Open WebStorm, create a new command-line-project.
2. Change the contents of main.dart to the following:
```
class Foo {
  final double foo;
  const Foo(this.foo);
}

const foo = const Foo(0.1337 * 42);

bar() {
  foo;
}

main(List<String> args) {
  bar();
}
```
3. Set a breakpoint at the function call inside main().
4. Run the debugger
5. When the breakpoint hits - Continue execution by 'stepping into' the bar function.

## What is the expected output? What do you see instead?
**Expected:**
The debugger steps into the bar function

**Actual:**
The vm crashes (windows notifies that dart.exe has stopped working).

## What version of the product are you using?
Webstorm: 11 EAP
DartSDK: 1.13.0-dev-2.0 (x64)

On what operating system?
Windows 7 (x64)

## Please provide any additional information below.
The issue has been reproduced using the 32bit sdk, and on mac with a previous version of webstorm (10.0.4).
The issue could not be reproduced using a previous version of the DartSDK.

**Comment by sethladd** [comment]
cc @danrubel do you think this is a JetBrains plugin issue or issue with the VM/legacy debug protocol?

**Comment by danrubel** [comment]
I doubt that it has anything to do with the protocol or the editor. The VM should not crash no matter what the client sends to it.

**Comment by johnmccutchan** [comment]
@turnidge @rmacnak-google

**Comment by iposva-google** [comment]
Running with a debug VM this points to the parser code:

/Volumes/Sources/iposva/dart-git/sdk/runtime/vm/parser.cc:3329: error: expected: func.NumParameters() == params.parameters->length()

**Comment by mhausner** [comment]
The VM dies when trying to compute the LocalVarDescriptors of the 'const foo' initializer expression to display the stack trace. This is a regression introduced by my recent change to eliminate the eager creation of var descriptors for execute-once code. I'll have a fix promptly.

**Comment by iposva-google** [comment]
Not shipped in 1.12. Removing the Priority-High label.

**Comment by mhausner** [comment]
https://codereview.chromium.org//1336363003
Fixed
Committed patchset #2 (id:20001) manually as
7720e0bed74562f901743bb999344434008cc33d (presubmit successful).
